src/main/webapp/app/communication/service/metis.service.ts

```diff
@@ -278,6 +278,17 @@ export class MetisService implements OnDestroy {
      * @return updated post
      */
     updatePost(post: Post): Observable<Post> {
+        if (post.id) {
+            const updateIndex = this.cachedPosts.findIndex((cachedPost) => cachedPost.id === post.id);
+
+            if (updateIndex !== -1) {
+                // We update the date immediately so that the client is aware about the post being edited without having to wait for the update call to finish
+                this.cachedPosts[updateIndex].updatedDate = dayjs();
+                this.cachedPosts[updateIndex].content = post.content;
+                this.posts$.next(this.cachedPosts);
+            }
+        }
```

Comment:
I think we can improve the code readability here

```suggestion
        if (post.id) {
            const updateIndex = this.cachedPosts.findIndex((cachedPost) => cachedPost.id === post.id);

            const foundCachedPost = updateIndex !== -1;
            if (foundCachedPost) {
                // We update the date immediately so that the client is aware about the post being edited without having to wait for the update call to finish
                this.cachedPosts[updateIndex].updatedDate = dayjs();
                this.cachedPosts[updateIndex].content = post.content;
                this.posts$.next(this.cachedPosts);
            }
        }
```

Reply:
Changed in this [commit](https://github.com/ls1intum/Artemis/pull/10799/commits/3acddde982e5e509c65dec3248df6791d9d56229).